pkg/subctl/cmd/diagnose/globalnet.go

```diff
@@ -0,0 +1,254 @@
+		return true
+	}
+
+	if cluster.Submariner.Spec.GlobalCIDR == "" {
+		status.Success("Globalnet is not installed - skipping")
+		return true
+	}
+
+	status.Start("Checking Globalnet configuration")
+	defer status.End()
+
+	retValue := checkClusterGlobalEgressIps(cluster, status) &&
+		checkGlobalEgressIps(cluster, status) &&
+		checkGlobalIngressIps(cluster, status)
+
+	if retValue {
+		status.EndWithSuccess("Globalnet is enabled and properly configured")
+	}
+
+	return retValue
+}
+
+func checkClusterGlobalEgressIps(cluster *cmd.Cluster, status reporter.Interface) bool {
+	clusterGlobalEgress, err := cluster.SubmClient.SubmarinerV1().ClusterGlobalEgressIPs(
+		corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.Warning("Error listing the ClusterGlobalEgressIP resources: %v", err)
+		return false
+	}
+
+	if len(clusterGlobalEgress.Items) != 1 {
+		status.Failure("Found %d ClusterGlobalEgressIP resources but only the default instance is supported", len(clusterGlobalEgress.Items))
+		return false
+	}
+
+	retValue := true
+
+	if clusterGlobalEgress.Items[0].Name != constants.ClusterGlobalEgressIPName {
+		status.Failure(
+			"The name of the ClusterGlobalEgressIP resource (%s) does not match the expected default name (%s)",
+			clusterGlobalEgress.Items[0].Name, constants.ClusterGlobalEgressIPName)
+
+		retValue = false
+	}
+
+	clusterGlobalEgressIP := clusterGlobalEgress.Items[0]
+
+	numberOfIPs := -1
+	if clusterGlobalEgressIP.Spec.NumberOfIPs != nil {
+		numberOfIPs = *clusterGlobalEgressIP.Spec.NumberOfIPs
+	}
+
+	if numberOfIPs != len(clusterGlobalEgressIP.Status.AllocatedIPs) {
+		status.Failure("The number of requested IPs (%d) does not match the number allocated (%d) for ClusterGlobalEgressIP %q",
+			numberOfIPs, len(clusterGlobalEgressIP.Status.AllocatedIPs), clusterGlobalEgressIP.Name)
+
+		retValue = false
+	}
+
+	condition := meta.FindStatusCondition(clusterGlobalEgressIP.Status.Conditions, string(submarinerv1.GlobalEgressIPAllocated))
+	if condition == nil {
+		status.Failure("ClusterGlobalEgressIP %q is missing the %q status condition", clusterGlobalEgressIP.Name,
+			submarinerv1.GlobalEgressIPAllocated)
+
+		retValue = false
+	} else if condition.Status != metav1.ConditionTrue {
+		status.Failure("The allocation of global IPs for ClusterGlobalEgressIP %q failed with reason %q and message %q",
+			clusterGlobalEgressIP.Name, condition.Reason, condition.Message)
+
+		retValue = false
+	}
+
+	return retValue
+}
+
+func checkGlobalEgressIps(cluster *cmd.Cluster, status reporter.Interface) bool {
+	globalEgressIps, err := cluster.SubmClient.SubmarinerV1().GlobalEgressIPs(
+		corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.Warning("Error obtaining GlobalEgressIPs resources: %v", err)
+		return false
+	}
+
+	retValue := true
+
+	for i := range globalEgressIps.Items {
+		gip := globalEgressIps.Items[i]
+		numberOfIPs := -1
```

Comment:
```suggestion
		numberOfIPs := 1
```

Reply:
Any special reason to set 1? 
I think -1 is a better option
since -1 is always != len(gip.Status.AllocatedIPs)

Reply:
We're defaulting it to 1. `gip.Spec.NumberOfIPs` is a pointer and thus can be nil if the user didn't set the field, which we don't require.

Reply:
Ack